common_types/src/record_batch.rs

```diff
@@ -269,27 +269,63 @@ impl RecordBatch {
     }
 }
 
+/// Converter from arrow record batches to inner's
+///
+/// Converting arrow schema to inner's is expensive. For eliminate the
+/// duplicated converting, we define such converter to cache the first converted
+/// one and reuse it in later converting.
+///
+/// NOTE: must ensure the converted arrow record batches has the same schema
+/// before deciding to use this converter.
+#[derive(Default)]
+pub struct CachedRecordBatchesConverter {
+    schema_cache: Option<RecordSchema>,
```

Comment:
How about init this field in `fn new()`, then `convert` could take `&self` directly.